service/src/main.rs

```diff
@@ -306,7 +309,14 @@ fn start_worker<E, T, D>(
 	let genesis_hash = node_api.genesis_hash.as_bytes().to_vec();
 
 	let tee_accountid = enclave_account(enclave.as_ref());
-	ensure_account_has_funds(&mut node_api, &tee_accountid);
+
+	if dev {
+		ensure_account_has_funds(&mut node_api, &tee_accountid);
+	} else if amount_to_be_funded(&mut node_api, &tee_accountid) > 0 {
```

Comment:
I think I don't get it - what is the logic behind this? If the amount to be funded is greater 0, we say we don't have enough funds?

Reply:
Yes and we stop the process.

Reply:
But why? That's what I don't understand 😛

Reply:
The amount_to_be_funded is the amount that the account is missing to have at least existential_deposit * MIN_FUND_INCREASE_FACTOR. i.e. about 1_000_000_000.
This amount is necessary for the account to be able to carry out transactions, such as registering the enclave.  
If the amount to be funded is greater 0, then we don't have enough funds. Registering the enclave will fail and this transaction will be banned for 30 mn. See 
https://github.com/integritee-network/worker/issues/538#issuecomment-987675269
I will add a comment.

Reply:
I agree with Felix here : We can have a lower value than `existential_deposit * MIN_FUND_INCREASE_FACTOR`, for example half of it (resulting in` amount_to_be_funded = 0.5 * existential_deposit * MIN_FUND_INCREASE_FACTOR`, which is greater than 0, and still have enough funds to carry out registrations. We just don't have the maximum amount available.

Reply:
I don't see why in case we don't call `ensure_account_has_funds`, we should allow a smaller amount. 
Until now, the method `ensure_account_has_funds` ensured that the enclave's account has initially `existential_deposit * MIN_FUND_INCREASE_FACTOR` . Why this value? It is not clear. See https://github.com/integritee-network/worker/issues/285#issuecomment-905509886
I have treated both cases in a similar way, as nothing else is required. I couldn't find out a better logic

Reply:
As you said : The exact value does not matter, as long as it's enough. So ` 0.5 * existential_deposit * MIN_FUND_INCREASE_FACTOR` might be enough as well, so we shouldn't return here. I think you should only return in case the enclave does not have:
a) enough funds to register it self. I don't know how much that costs or how you could find that out. In case it requires a lot of extra implentation, lets go for option b:
b) Just check for `existential_deposit`. If it's below that, the enclave can't do anything, so simply return.

Reply:
Ah okay, thanks for the explanation. So maybe we should name our functions differently - or introduce a new function that is named something like `does_account_have_sufficient_funds` ? Otherwise it's confusing to read? Or do you disagree?

Reply:
b) will not work. As Registering the enclave will fail and this transaction will be banned for 30 mn.
for a) I have to check how.

Reply:
Does it matter if it's banned for 30 min? It's not dev mode, but production, where enough funds should be ensured anyway. And if the account does already exist (otherwise it will be below the ED, I think?) it should have been fauceted in some way.

Reply:
We discussed it : See https://github.com/integritee-network/worker/issues/538#issuecomment-987675269

Reply:
> Ah okay, thanks for the explanation. So maybe we should name our functions differently - or introduce a new function that is named something like `does_account_have_sufficient_funds` ? Otherwise it's confusing to read? Or do you disagree?

Reply:
A method `does_account_have_sufficient_funds` would do the same as `amount_to_be_funded`, except it would return a bool. I would prefer to rename `amount_to_be_funded`. For me it sounds clear, but ... What about `missing_funds_amount`

Reply:
Well if it's clear for everybody else except me, then we leave it as is ☺️